spec/wcc/contentful_spec.rb

```diff
@@ -30,6 +32,96 @@
     WCC::Contentful::Model.class_variable_get('@@registry').clear
   end
 
+  describe '.init with preview token' do
+    context 'with preview token' do
+      before do
+        WCC::Contentful.configure do |config|
+          config.access_token = valid_contentful_access_token
+          config.space = valid_contentful_space_id
+          config.store = nil
+          config.preview_token = valid_contentful_preview_token
+          config.content_delivery = :direct
+        end
+      end
+
+      it 'should populate models via Preview client' do
+        # act
+        VCR.use_cassette(
+          'WCC_Contentful/_init/with_preview_token/init_with_preview_token',
+          record: :none
+        ) do
+          WCC::Contentful.init!
+
+          # assert
+          content_type = WCC::Contentful::Model::Redirect.content_type
+          expect(content_type).to eq('redirect')
+        end
+      end
+
+      it 'should find published content in Contentful' do
+        # act
+        VCR.use_cassette(
+          'WCC_Contentful/_init/with_preview_token/init_with_preview_token',
+          record: :none
+        ) do
+          WCC::Contentful.init!
+
+          VCR.use_cassette(
+            'WCC_Contentful/_init/with_preview_token/published_redirect',
+            record: :none
+          ) do
+            redirect = WCC::Contentful::Model::Redirect.find_by(slug: 'published-redirect')
+
+            expect(redirect).to_not be_nil
+            expect(redirect.url).to eq('https://watermark.formstack.com/forms/theporch')
+          end
+        end
+      end
+
+      it 'should not find draft content in Contentful if no preview password is given' do
```

Comment:
This is still the case. I thought you said the other day that the gem shouldn't care about the password? Because of that, I just made it to where if it the gem receives a preview param at all, it will use the preview client, but on the porch side I'm doing a check for the password, and if it's correct I do a call that includes the preview param.